bridge/slack/slack.go

```diff
@@ -356,6 +356,10 @@ func (b *Bslack) editMessage(msg *config.Message, channelInfo *slack.Channel) (b
 }
 
 func (b *Bslack) postMessage(msg *config.Message, messageParameters *slack.PostMessageParameters, channelInfo *slack.Channel) (string, error) {
+	// don't post empty messages
+	if msg.Text == "" {
```

Comment:
This doesn't look right to me. What's the rationale behind this? If it's simply a file share than there might not be any text originally associated with it.

Also, if we do expect to always have text on any message then we should ensure that in all the inbound message pipelines on the bridges.

In the case of Slack's inbound pipeline we already check for wellformed messages [with the following](https://github.com/42wim/matterbridge/blob/c6fd65d1d7f5085890c0a00d37be8d6abb458921/bridge/slack/handlers.go#L181-L190):
```go
	if len(ev.Files) == 0 && (rmsg.Text == "" || rmsg.Username == "") {
		if ev.BotID != "" {
			// This is probably a webhook we couldn't resolve.
			return nil, fmt.Errorf("message handling resulted in an empty bot message (probably an incoming webhook we couldn't resolve): %#v", ev)
		}
		return nil, fmt.Errorf("message handling resulted in an empty message: %#v", ev)
	}
```
This would tend to say that we most definitely accept messages with no text as long as they contain files.

Reply:
The rationale is to fix https://github.com/42wim/matterbridge/issues/574
Yes, we accept messages with no text as they contain files, but those file messages get handled by `uploadFile` (which handles the file and file comment) before calling `postMessage`.  This means that `postMessage` will just post an empty line (if the text is empty, eg by slack regardless of we have a file comment) which we don't want.